src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -294,7 +290,7 @@ def deploy_nsd_from_bicep(
             os.path.join(self.config.build_output_folder_name, NF_DEFINITION_BICEP_FILE)
         )
 
-        with open(self.config.arm_template.file_path, "w") as file:
+        with open(self.config.arm_template.file_path, "w", encoding="utf-8") as file:
```

Comment:
I don't think adding encoding changes anything here but it is a little bit of a change in logic so you might want to test